app/actions/post.go

```diff
@@ -21,12 +21,39 @@ import (
 type CreateNewPost struct {
 	Title       string             `json:"title"`
 	Description string             `json:"description"`
+	TagSlugs    []string           `json:"tags"`
 	Attachments []*dto.ImageUpload `json:"attachments"`
+
+	Tags []*entity.Tag
+}
+
+// OnPreExecute prefetches Post for later use
+func (input *CreateNewPost) OnPreExecute(ctx context.Context) error {
+	input.Tags = make([]*entity.Tag, len(input.TagSlugs))
+	for i, slug := range input.TagSlugs {
+		getTag := &query.GetTagBySlug{Slug: slug}
+		if err := bus.Dispatch(ctx, getTag); err != nil {
+			return err
+		}
+		
+		input.Tags[i] = getTag.Result
+	}
+	
+	return nil
 }
```

Comment:
Good idea to make this part of OnPreExecute. You'll just need to update that comment though.